packages/bigtable/src/row.js

```diff
@@ -158,25 +123,58 @@ util.inherits(Row, common.GrpcServiceObject);
  * // }
  */
 Row.formatChunks_ = function(chunks) {
-  var families = [];
-  var chunkList = [];
+  var rows = [];
+  var familyName;
+  var qualifierName;
+
+  chunks.reduce(function(row, chunk) {
+    var family;
+    var qualifier;
+
+    row.data = row.data || {};
 
-  chunks.forEach(function(chunk) {
-    if (chunk.resetRow) {
-      chunkList = [];
+    if (chunk.rowKey) {
+      row.key = Mutation.convertFromBytes(chunk.rowKey);
     }
 
-    if (chunk.rowContents) {
-      chunkList.push(chunk.rowContents);
+    if (chunk.familyName) {
+      familyName = chunk.familyName.value;
+    }
+
+    if (familyName) {
+      family = row.data[familyName] = row.data[familyName] || {};
```

Comment:
In your test above you might see a false positive because you do the `familyDoubleAssignment` before you do `familyExtend`. That would create the reference and then extend it.